src/OpenTelemetry.Instrumentation.EventCounters/EventCountersMetrics.cs

```diff
@@ -132,7 +192,7 @@ private void UpdateInstrumentWithEvent(bool isGauge, string eventSourceName, str
             ValueTuple<string, string> metricKey = new(eventSourceName, name);
             _ = this.values.AddOrUpdate(metricKey, value, isGauge ? (_, _) => value : (_, existing) => existing + value);
 
-            var instrumentName = $"EventCounters.{eventSourceName}.{name}";
+            var instrumentName = GetInstrumentName(eventSourceName, name);
```

Comment:
Move this inside the `if` check below to only call `GetInstrumentName` method when `instruments` does not contain the given `metricKey`.